src/pages/SearchPage.js

```diff
@@ -101,48 +111,112 @@ class SearchPage extends Component {
         }];
 
         if (this.state.userToInvite) {
-            sections.push(({
+            sections.push({
                 undefined,
                 data: [this.state.userToInvite],
                 shouldShow: true,
                 indexOffset: 0,
-            }));
+            });
         }
 
         return sections;
     }
 
     /**
-     * Reset the search value and redirect to the selected report
-     *
-     * @param {Object} option
-     */
+   * Reset the search value and redirect to the selected report
+   * @param {Object} option
+   */
     selectReport(option) {
         if (!option) {
             return;
         }
 
         if (option.reportID) {
+            this.setState(
+                {
+                    searchValue: '',
+                },
+                () => {
+                    Navigation.navigate(ROUTES.getReportRoute(option.reportID));
+                },
+            );
+        } else {
+            fetchOrCreateChatReport([this.props.session.email, option.login]);
+        }
+    }
+
+
+    /**
+     * Validates search input via regexes and validation API (phone numbers only)
+     * @param {String} searchValue
+     * @returns {void}
+     */
+    validateInput(searchValue) {
```

Comment:
I believe we're going to have to move this into `OptionsListUtils` since the validation you're adding is going to apply to **all** pages that have account-search functionality. Can you figure out how we can reuse `getSearchOptions` here (by editing `getOptions`)? I'm thinking we can have `getOptions` also return `headerMessage`

Reply:
I discussed this with you earlier. That my initial approach was to add the changes in `getOptions` only but it's not flexible enough and might require good amount of refactoring.

Reply:
Woah you did? I'm sorry I don't remember you bringing up that approach, maybe I misunderstood :O

Hmm I don't see why we can't refactor `getHeaderMessage` just a bit, and make `getOptions` also return something like `headerMessage: getMeaderMessage(...)`. Here's why I'm thinking this will work (please let me know if you disagree on these points):
1. `getHeaderMessage` takes only a few parameters, and all of the params should be able to be calculated from other variables inside `getOptions`
2. `getOptions` is only used in `OptionsListUtils`, which has `get...Options` functions for specific pages, all of those pages will need this type of functionality (the same type of logic for `headerMessage`)

Reply:
Hi @Beamanator 
I tried merging these changes into utils. So `getSearchOptions` & `getHeaderMessage` are not connect. They both take the `searchValue` as one of the input. With this, we might end up hitting the phone number validation API twice.

Reply:
Hmm once you push your latest changes I can check and see if I can help come up with a way to only hit that validation API once 👍